bundle/config/resources.go

```diff
@@ -126,6 +126,46 @@ func (r *Resources) VerifyUniqueResourceIdentifiers() (*UniqueResourceIdTracker,
 	return tracker, nil
 }
 
+func (r *Resources) VerifyAllResourcesDefined() error {
+	for k, e := range r.Jobs {
+		if e == nil || !e.DynamicValue.IsValid() {
+			return fmt.Errorf("job %s is not defined", k)
```

Comment:
Is it possible to use `diag` and return the location where the undefined job is located?

Reply:
I would prefer enforcing this is implemented by adding a function to `ConfigResource`.

Then all new resources have to implement the function or it won't compile.

Reply:
+1 we need to increasingly think about how to make it really simple to add new ConfigResources

Reply:
@shreyas-goenka we can but location won't be available because `e.DynamicValue` is not valid